[Feat] 장르별 레퍼런스 탐색 기능 생성

code-Lance opened the pull request:
## 🔗 관련 이슈
해당 PR이 연결되는 이슈 번호를 명시하세요.  
예) Issue #1 

---

## ✨ 변경 내용
이번 PR에서 어떤 작업을 했는지 요약해주세요.
- 장르별 노래 랭킹 기능 생성
- 유튜브 링크 조회 기능 생성

---

## 🧪 테스트 방법
어떻게 동작을 검증했는지 작성해주세요.
- [ ] Swagger을 통한 장르별 랭킹 조회 확인
- [ ] 앨범 이미지를 잘 가져오는 지 확인
- [ ] 오류 메시지를 설정한 대로 잘 가져오는 지 확인
- [ ] youtube 링크를 잘 가져오는지 확인

---

## 👀 리뷰 포인트
리뷰어가 집중해서 봐줬으면 하는 부분을 적어주세요.
- API 명세서에 유튜브 링크를 제공하는 API가 누락되어 있었습니다. 일단 구현을 하긴 했지만 확인 부탁드립니다.
- 임시로 추가한 SecurityConfig 확인 부탁드립니다.
- 스웨거와 id 'org.springframework.boot' 버전차이로 인해 version '3.3.5 로 변경하였으니 괜찮은지 확인 부탁드립니다.
---

## 📎 기타 참고 사항
### 장르별 노래 레퍼런스 제공 기능 사진
- 스웨거
<img width="1411" height="880" alt="image" src="https://github.com/user-attachments/assets/2795677f-ce67-42f6-9937-07e6ce4200fe" />

- 앨범 이미지 확인 (첫번째 노래)
<img width="1728" height="946" alt="image" src="https://github.com/user-attachments/assets/1e4178f5-af5c-4112-ad03-b44ac402694e" />

- 오류 코드 확인
<img width="460" height="211" alt="image" src="https://github.com/user-attachments/assets/b1834e6f-b281-40a8-a769-37dc6b52b259" />

### 유튜브 링크 제공 기능 
- 스웨거
<img width="1402" height="651" alt="image" src="https://github.com/user-attachments/assets/c7a7171d-3de9-468e-88a2-5b3f21f4eb31" />

- 유튜브 링크 확인
<img width="1728" height="948" alt="image" src="https://github.com/user-attachments/assets/74fe9e2e-91c0-434e-a063-3aa935dbeb66" />

- 오류 확인
<img width="566" height="230" alt="image" src="https://github.com/user-attachments/assets/966f920c-9408-4752-8fe1-8b45c964da83" />

code-Lance committed:
[Feat] 장르별 레퍼런스 탐색 기능 생성
1. 장르별 노래 랭킹 기능 생성
- spotify api를 활용하여 조회
- 노래 제목, 아티스트, 앨범 이미지 주소 제공
2. 유튜브 링크 조회 기능 생성
- 노래 제목, 아티스트 명을 입력받고 youtube data v3 api를 통해 조회

code-Lance committed:
[Docs] ReferenceController 누락된 @NotBlank 메시지 추가
1. ReferenceController 누락된 @NotBlank 메시지 추가
- 아티스트 이름 누락 관련 메시지 추가

seominjae1 commented on src/main/java/com/fifo/compasstep/reference/exceptions/handler/ReferenceExceptionAdvice.java:
```diff
@@ -0,0 +1,77 @@
+        }
+        if ("title".equals(name)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(ApiResponse.onFailure(400, "곡 제목을 지정해주세요.", null));
```
오류 코드는 한곳에서 보기 쉽게 ReferenceErrorStatus.java에 같아두는게 어떨까요?

seominjae1 commented on src/main/java/com/fifo/compasstep/config/SecurityConfig.java:
SecurityConfig는 제꺼랑 달라서 제가 추가할까요?

code-Lance replied:
그러면 좋을 것 같습니다

seominjae1 commented on src/main/java/com/fifo/compasstep/reference/exceptions/handler/ReferenceExceptionAdvice.java:
별다른 이유가 없다면 apiPayload/exceptions/ExceptionAdvice에 통합하는게 어떤가요?
@Order로 순위를 일일이 지정하는 것보다 하나로 통합하면 더 관리하기 편할 것 같아서 말씀드립니다!

seominjae1 replied:
+apiPayload에 ExceptionsHandler 추가해야 할 것 같아요!

seominjae1 commented on src/main/java/com/fifo/compasstep/reference/exceptions/handler/ReferenceExceptionAdvice.java:
```diff
@@ -0,0 +1,77 @@
+        }
+        if ("artist".equals(name)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body(ApiResponse.onFailure(400, "아티스트를 지정해주세요.", null));
```
여기도요

seominjae1 reviewed (commented):
수고하셨습니다!
코멘트 확인해주세요!

seominjae1 commented on src/main/java/com/fifo/compasstep/reference/controller/ReferenceController.java:
```diff
@@ -0,0 +1,51 @@
+        List<RankingItemDto> list = genreRankingService.getGenreRanking(
+                new ReferenceRequestDto(genre, market, limit)
+        );
+        return new ApiResponse<>(200, "장르별 랭킹 조회를 성공했습니다.", list);
```
저희 success도 따로 메세지를 분리하나요?

code-Lance replied:
일단 저희 API 명세서에는 저렇게 나눠져 있어서 분리하긴 했습니다

code-Lance committed:
[Refactor] 장르별 레퍼런스 탐색 기능 PR 기반 수정
1. 오류 코드 ReferenceErrorStatus 에 두기
- 400, "곡 제목을 지정해주세요."
- 400, "아티스트를 지정해주세요."
2. ReferenceExceptionAdvice를 ExceptionAdvice에 통합
- ExceptionAdvice에 통합
- ReferenceExceptionAdvice 파일 삭제
